services/src/test/scala/cromwell/pipeline/service/ProjectServiceTestImpl.scala

```diff
@@ -0,0 +1,54 @@
+package cromwell.pipeline.service
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.model.wrapper.UserId
+
+import scala.concurrent.Future
+
+class ProjectServiceTestImpl(project: Project, testMode: TestMode) extends ProjectService {
+
+  private[service] def getUserProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(project)
+    }
+
+  def getUserProjectByName(namePattern: String, userId: UserId): Future[Project] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(project)
+    }
+
+  def addProject(request: ProjectAdditionRequest, userId: UserId): Future[Either[VersioningException, Project]] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(Right(project))
```

Comment:
Unfortunately this part won't be that easy. I think it this case you will have to pattern match on error type like this.
```suggestion
      case WithException(exc: VersioningException) => Future.successful(Left(exc))
      case WithException(exc)                      => Future.failed(exc)
      case _                                       => Future.successful(Right(project))
```
Doesn't look great, but I think it will imitate expected behavior better.

Reply:
Fixed